### PR TITLE
Adding in RSVP buttons and notifications.

### PR DESCRIPTION
**Pull request by JnBrymn**
# What
Added RSVP buttons that will create a Participant for the user and penny_chat with the appropriate role. It will also notify the organizer.

# Why
Richer functionality so that we don't have to rely on external tools for setting up meetings.

# Relates to
https://github.com/penny-university/penny_university/issues/117

### CODE REVIEW

**Comment by JnBrymn** [comment]
# Evidence of completeness in Dev

## Inviting a user
1. Create temp email and new slack user with that email.
2. Created penny chat
![image](https://user-images.githubusercontent.com/566533/74612275-77655680-50c9-11ea-92d2-63192dec696b.png)
3. Reviewed and chose to share it 
![image](https://user-images.githubusercontent.com/566533/74612283-8f3cda80-50c9-11ea-84f0-3ea379d042fb.png)
4. Invitee got an invite and indicated they can't come
![image](https://user-images.githubusercontent.com/566533/74612293-b72c3e00-50c9-11ea-80b0-d76d11271126.png)
5. Organizer got un-RSVP
![image](https://user-images.githubusercontent.com/566533/74612308-da56ed80-50c9-11ea-86a4-fe1549d30747.png)
6. Invitee then indicated they can come and Organizer got rsvp
![image](https://user-images.githubusercontent.com/566533/74612318-f5296200-50c9-11ea-956b-1d1e48689d60.png)
7. Invitee then indicated AGAIN they can come and Organizer was not reminded again (as expected)

Then I checked the database
```python
In [3]: ps = list(Participant.objects.filter(penny_chat__title='test title'))                                                                                                                                                                 
In [5]: ps[0].user.display_name                                                                                                                                                                                                               
Out[5]: 'Turd Fergison'
In [6]: ps[0].role                                                                                                                                                                                                                            
Out[6]: 20
In [7]: Participant.ROLE_CHOICES                                                                                                                                                                                                              
Out[7]: 
((10, 'Organizer'),
 (20, 'Attendee'),
 (30, 'Invitee'),
 (40, 'Invited NonAttendee'))

In [11]: ps[1].user.email                                                                                                                                                                                                                     
Out[11]: 'jfberryman@gmail.com'
In [8]: ps[1].role                                                                                                                                                                                                                            
Out[8]: 10
```
All as expected.

## Inviting a channel

File: bot/processors/pennychat.py [hunk 1]
```diff
@@ -477,24 +547,113 @@ def share(self, event):
...
+        event is handled here.
+
+        There are two side effects of this method:
+         1. A Participant entry for the penny_chat and the user will be created or updated with the appropriate role.
+         2. The organizer will be notified of "important" changes.
+         3. The user will be told that the organizer will be notified.
+
+         The specifics are rather complicated, but you can see how they work in
+         bot.tests.processors.test_pennychat.test_PennyChatBotModule_attendance_selection
+
```
Comment (nchouard) [inline bot/processors/pennychat.py]: This is teaching me to write better docs. Thank you!

File: bot/tests/processors/test_pennychat.py [hunk 2]
```diff
@@ -150,3 +165,177 @@ def id_mock(user_id, slack_client):
...
+        starting_role=None,
+        can_attend=False,
+        expected_final_role=None,
+        expected_organizer_notified=False,
+    ),
+]
+
+
+@pytest.mark.django_db
+@pytest.mark.parametrize(",".join(testdata[0].keys()), [td.values() for td in testdata])
```
Comment (nchouard) [inline bot/tests/processors/test_pennychat.py]: Ok, this is awesome. Wasn't aware of this before!

File: pennychat/models.py [hunk 3]
```diff
@@ -66,13 +67,15 @@ def __repr__(self):
 
 
 class Participant(models.Model):
-    ORGANIZER = 1
-    ATTENDEE = 2
-    INVITEE = 3
+    ORGANIZER = 10
+    ATTENDEE = 20  # Not necessarily invited by name, could have come from a channel invite
```
Comment (nchouard) [inline pennychat/models.py]: For some reason, this comment is confusing to me. I get what you are saying, but it makes it seem like "ATTENDEE" is some sort of other form of "INVITEE", when really it's just "anyone who has _accepted_ RSVPed". Low FG score for actually changing the comment, but I figured I'd let you know how I read it (and how a future fellow coder may read it? 😄).

File: bot/tests/processors/test_pennychat.py (hunk 2 above)
Comment (JnBrymn) [reply bot/tests/processors/test_pennychat.py]: yeah! me neither - super useful

File: pennychat/models.py (hunk 3 above)
Comment (JnBrymn) [reply pennychat/models.py]: I updated it a bit.

**Pull request merged**